build-tools/xa-prep-tasks/Xamarin.Android.BuildTools.PrepTasks/JdkInfo.cs

```diff
@@ -26,32 +27,30 @@ public class JdkInfo : Task
 
 		public override bool Execute ()
 		{
-			Log.LogMessage (MessageImportance.Low, $"Task {nameof (JdkInfo)}");
-			Log.LogMessage (MessageImportance.Low, $"  {nameof (Output)}: {Output}");
-			Log.LogMessage (MessageImportance.Low, $"  {nameof (AndroidNdkPath)}: {AndroidNdkPath}");
-			Log.LogMessage (MessageImportance.Low, $"  {nameof (AndroidSdkPath)}: {AndroidSdkPath}");
-			Log.LogMessage (MessageImportance.Low, $"  {nameof (JavaSdkPath)}: {JavaSdkPath}");
-
-			var androidSdk    = new AndroidSdkInfo (CreateTaskLogger (this), AndroidSdkPath, AndroidNdkPath, JavaSdkPath);
+			var logger        = CreateTaskLogger (this);
+			var androidSdk    = new AndroidSdkInfo (logger, AndroidSdkPath, AndroidNdkPath, JavaSdkPath);
 			try {
-				var javaSdkPath = androidSdk.JavaSdkPath;
-				if (string.IsNullOrEmpty(javaSdkPath)) {
+				if (string.IsNullOrEmpty(androidSdk.JavaSdkPath)) {
 					Log.LogError ("JavaSdkPath is blank");
 					return false;
 				}
 
-				Log.LogMessage (MessageImportance.Low, $"  {nameof (androidSdk.JavaSdkPath)}: {javaSdkPath}");
+				Log.LogMessage (MessageImportance.Low, $"  {nameof (androidSdk.JavaSdkPath)}: {androidSdk.JavaSdkPath}");
+
+				var maxVersion = new Version ("8.0");
+				Tools.JdkInfo jdk = Tools.JdkInfo.GetKnownSystemJdkInfos (logger)
```

Comment:
This causes us to *completely ignore* the `androidSdk.JavaSdkPath` value, meaning `$(JavaSdkDirectory)` is now entirely ignored.

This seems...wrong? :-)

Instead, I *think* we want (-ish):

```csharp
var jdks = new[]{new JdkInfo (androidSdk.JavaSdkPath)}.Concat (Tools.JdkInfo.GetKnownSystemJdkInfos (logger));
var jdk = jdks.Where (j => j.MaxVersion != null ? j.Version <= maxVersion : true)
    .Where (j => j.IncludePath.Any ())
    .FirstOrDefault ();
if (jdk == null)
    Log.LogError (…);
// use jdk...
```

Error checking excluded.  If `$(JavaSdkDirectory)` is set and is valid -- i.e. `AndroidSdkInfo` was constructed -- then we should use `new JdkInfo ($(JavaSdkDirectory))` aka `new JdkInfo (androidSdk.JavaSdkPath)`, and use that *preferentially*.

Question about this preferential use: if e.g. JDK 9 is specified, *should we use it*?  That might be helpful for testing new JDK versions.

If `$(JavaSdkDirectory)` *isn't* set/`AndroidSdkInfo` doesn't construct, *then* we should consult `JdkInfo.GetKnownSystemJdkInfos()`.